st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -277,17 +294,35 @@ def validate(self):
 
             return self
 
-    @jsexpose(body_cls=ExecutionSpecificationAPI, status_code=http_client.CREATED)
-    def post(self, spec_api, execution_id, no_merge=False):
+    def post(self, spec_api, id, requester_user=None, no_merge=False, **kwargs):
         """
         Re-run the provided action execution optionally specifying override parameters.
 
         Handles requests:
 
             POST /executions/<id>/re_run
         """
+
+        if (spec_api.tasks or spec_api.reset) and spec_api.parameters:
```

Comment:
It seems like you moved this outside API model `.validate()` method to the controller method?